[IMP] General: Moving product catalog page and per unit pricing (#18240)
odoo/documentation: content/applications/essentials/product_catalog.rst

===============
Product catalog
===============

.. |SO| replace:: :abbr:`SO (sales order)`
.. |SOs| replace:: :abbr:`SOs (sales orders)`
.. |BoM| replace:: :abbr:`BoM (bill of materials)`
.. |RfQ| replace:: :abbr:`RfQ (request for quotation)`

The *product catalog* is a feature integrated with any Odoo app that allows users to add products or
components to an order. This includes the **Inventory**, **Manufacturing**, **Sales**, **Purchase**,

> and **Repairs** apps, among others.

Reviewer: ```suggestion
The Odoo *product catalog* allows users to add products or
components to a record in the **Inventory**, **Manufacturing**, **Sales**, **Purchase**,
and **Repairs** apps, among others.
```

The product catalog simplifies the creation of new sales orders (SOs), purchase orders (POs),
manufacturing orders (MOs), bill of materials (BoMs), and more, by providing a visual interface

> through which products and components can be quickly selected.

Reviewer: consider a bulleted list, i.e.:

> By providing a visual interface through which products and components can be quickly selected, the product catalog simplifies the creation of database records including:
> - SOs
> - POs
> - ... etc.

Reviewer: also include quotations/RFQ form, and remove those from the next paragraph

Reviewer: also also: consider linking to the respective articles for these record types

The product catalog can be accessed from the first tab of a quotation, request for quotation (RfQ),
order, or bill of materials (BoM) form, and opens in a new page when selected. Products and
components in the catalog are displayed in a user-friendly, POS-style format. They from can be

> selected from the catalog and added to forms.

Reviewer: ```suggestion
The product catalog can be accessed from the first tab of the record form, and opens in a new page when selected. Products and
components in the catalog are displayed in a user-friendly, point of sale-style format. They can be
selected from the catalog and added to forms.
```

Reviewer: consider using `abbr` if you don't want to spell out POS

Author: Probably worth doing for consistency with other abbreviations. Good note!

> Using the product catalog

Reviewer: ```suggestion
View the product catalog
```

Reviewer: prefer present/imperative over gerund
consider renaming sections for specificity

Author: Man, I love me some gerunds. I always do that.

=========================

> To use the product catalog, begin by creating or opening a quotation, |RfQ|, order, or |BoM|. This

Reviewer: ```suggestion
To view the product catalog, create or open a quotation, |RfQ|, order, or |BoM|. This
```

> example uses a sales quotation. The UI may differ from what is pictured depending on the apps and

Reviewer: readers could miss this single sentence, consider placing screenshots in `example` admonition blocks to reiterate that the article just covers one type of record

modules active on the Odoo database.

On the quotation form, fill out the :guilabel:`Customer` field. Be sure to fill out the
:guilabel:`Invoice Address` and :guilabel:`Delivery Address` fields of those are available. Products

> cannot be added to the form until this essential information has been entered.

Reviewer: consider specifying the required fields for other form types, too

Reviewer: eta: tested in master runbot and _catalog_ cannot be viewed before filling out required fields, but _products_ CAN be added

Author: I didn't even think to test that. Weird!

Click the :guilabel:`Catalog` link to open the catalog in a new page.

.. image:: product_catalog/catalog-button.png

>    :alt: The "Catalog" button on the "Order Lines" tab of a sales quotation.

Reviewer: in master runbot quotation form, i'm seeing _quote builder_ tab instead of _optional products_, and no **description** column header
<img width="1365" height="259" alt="Image" src="https://github.com/user-attachments/assets/b9555f24-b1c3-4226-bba3-a7d0e580acd8" />

The product catalog displays a card for each product added to Odoo. Each card displays a few key
details about the corresponding product:

- Product photo
- Product title
- Price or cost of the product, depending on whether it is bought, sold, or used as a component
- Reference code (e.g. *DESK0005*)
- On-hand quantity

> - Variant attributes (e.g., colors, materials, customizaztions, etc.)

Reviewer: possibly for discussion with felicia & team: should we be adding periods for a bulleted list of items without em dash description as well?

.. image:: product_catalog/product-card.png

>    :alt: A product card in the product catalog.

Reviewer: checked in master and it seems like catalog product cards
1. have **add** button
2. variants are displayed separately with a pill specifying the variant

not sure if that's an adjustable setting? if not, update screenshot

<img width="2557" height="1398" alt="Image" src="https://github.com/user-attachments/assets/bacef453-3fc9-46c8-8e21-5cc51afa1eaa" />

Author: Ah, interesting. The current screenshot is from accessing the product catalog on its own, not from getting there via a quotation. I'll update it to reflect that flow.

> To add a product to a form, click on the product's card or the :icon:`fa-shopping-cart`

Reviewer: ```suggestion
Add or remove a product
-----------------------

To add a product to a form, click on the product's card or the :icon:`fa-shopping-cart`
```

Reviewer: if you do accept my section structure suggestion, this should be the last section since it's the final action in the standard workflow (comes after viewing/searching/filtering)

Reviewer: could also separate add/remove actions into separate subsections, if you like

:guilabel:`Add` button in the bottom-right corner of the card. Doing so adds one unit of the

> product, which is displayed in a field in the bottom-left corner of the card.

Reviewer: consider removing left/right language, to accommodate RTL languages
i.e. `bottom corner` only instead of `bottom-right/bottom-left`

To adjust the quantity of the product added, click the :icon:`fa-minus` :guilabel:`(minus)` and
:icon:`fa-plus` :guilabel:`(plus)` buttons to adjust the quantity by one. Alternatively, a specific
quantity can be entered by typing it in manually. Clicking the product card additional times also
increases the product count with each click.

To remove a product from the order, either click the :icon:`fa-trash` :guilabel:`Remove` button in

> the bottom-right corner of the product card or click the :icon:`fa-minus` :guilabel:`(minus)` button

Reviewer: remove "right" for RTL

until the quantity is zero.

Once the desired quantity of each product has been reached, return to the form by clicking the
:guilabel:`Back to [X]` button at the top of the screen. This button differs depending on the type
of form being configured (quotation, |BoM|, etc.).

.. important::
   Products appear in the product catalog and can be added to orders even if there are zero units of
   the product on hand. As a result, it is important to confirm the quantity of a product being
   added to an order is actually available.

> Products with packaging options

Reviewer: if you do accept my section structure suggestion, this could be a subsection or admonition block under _add/remove a product_ since it's related to quantity/cost

-------------------------------

For products with multiple packaging options, a drop-down menu appears when that product is
selected. Choosing a packaging option other than the default displays both the cost for that package
and the per-unit cost for the items within it.

.. image:: product_catalog/per-unit-dropdown.png
   :alt: A product card with the packaging drop-down expanded and showing per-unit pricing.

> Product catalog filters

Reviewer: ```suggestion
Use search filters
```

Reviewer: if you do accept my section structure suggestion, this would be the second section
consider using an imperative for top-level heading

-----------------------

The product catalog can be filtered using the search bar at the top of the page or the sidebar on

> the left side of the page.

Reviewer: remove "left" for RTL

To filter by product type, click the :icon:`fa-caret-down` :guilabel:`(Toggle Search Panel)` button

> on the right side of the search bar to open the search menu. In the :icon:`fa-filter`

Reviewer: remove "right" for RTL

:guilabel:`Filters` section, select the :guilabel:`Services` filter to only show service products or
the :guilabel:`Goods` filter to only show physical products.

When creating or configuring a quotation, an :guilabel:`In the Order` filter appears in the
:icon:`fa-filter` :guilabel:`Filters` section of the search bar. Select this filter to only show
products that have already been added to the form.

> In the sidebar on the left side of the page, select an option in the :icon:`fa-th-list`

Reviewer: remove "on the left side of the page" for RTL

The Odoo *product catalog* allows users to add products or components to a record in the
**Inventory**, **Manufacturing**, **Sales**, **Purchase**, and **Repairs** apps, among others.

The product catalog simplifies the creation of new:

- :doc:`Sales quotations and orders (SOs) <../sales/sales/sales_quotations/create_quotations>`
- Purchase orders (POs)
- :doc:`Manufacturing orders (MOs)
  <../inventory_and_mrp/manufacturing/basic_setup/manufacturing_work_orders>`
- :doc:`Bills of materials (BoMs)
  <../inventory_and_mrp/manufacturing/basic_setup/bill_configuration>`
- :doc:`Requests for quotation (RFQs) <../inventory_and_mrp/purchase/manage_deals/rfq>`
- And more.

The product catalog provides a visual interface through which products and components can be quickly
selected. It can be accessed from the first tab of the record form, and opens in a new page when
selected. Products and components in the catalog are displayed in a user-friendly, point of
sale-style format. They can be selected from the catalog and added to forms.

Access and use the product catalog
==================================

Access the catalog directly
---------------------------

> The Product Catalog can be accessed directly by opening a supported app (such as **Sales**,

Reviewer 2: ```suggestion
The product catalog can be accessed directly by opening a supported app (such as **Sales**,
```

**Purchase**, **Manufacturing**, etc.) Inside a supported app, go to :menuselection:`Products -->
Products` to view the product catalog with a filter active for that app.

.. image:: product_catalog/catalog-purchase-example.png
   :alt: The product catalog as accessed through the **Purchase** app.

Access the catalog through a database record
--------------------------------------------

To view the product catalog through a database record in a supported app (such as **Sales**,
**Purchase**, **Manufacturing**, etc.), create or open a sales quotation or |SO|, |RfQ|, |MO|, or

> |BoM|. On the record form, click :guilabel:`Catalog` under the Product line. The required fields

Reviewer 2: ```suggestion
|BoM|. On the record form, click :guilabel:`Catalog` under the product line. The required fields
```
getting nitpicky here but im wondering which capitalization or formatting we should use here!

"product line" is an Odoo-specific term that refers to a line item  in the Products tab. Because of that i don't believe it's a proper noun and warrants a capitalization here! There isn't a hard rule, but something we've been doing in practice as a team is italicizing odoo-specific terms.

Do you think a beginner user would know what the product line is? If we were to go the super specific route, we might say instead: "click :guilabel:`Catalog` under the lines the *Products* tab". Not sure about the level of specificity we should get into here. 

TLDR: 
- i think we should un-capitalize "product catalog" but 
- letting you decide whether we leave it as is or italicize it
- or err on the side of specificity and describe the location instead of using the odoo-specific term!

Author: Hm... I think erring on the side of specificity is probably the way to go. Because "a product line" and "the product line" are so syntactically similar but semantically different, I think it's worth being as specific as possible. I'll revise with that in mind. Good feedback!
